src/classes/SftpSessionHandler.ts

```diff
@@ -401,74 +497,162 @@ export class SftpSessionHandler {
-              'Response: Status (OK)',
-              {
-                reqId,
-                code: SFTP_STATUS_CODE.OK,
-                path: temporaryFile.virtualPath,
-              },
-            );
-            this.sftpConnection.status(reqId, SFTP_STATUS_CODE.OK);
-          }).catch((err) => {
-            logger.debug(err);
-            logger.verbose(
-              'Response: Status (FAILURE)',
-              {
-                reqId,
-                code: SFTP_STATUS_CODE.FAILURE,
-                path: temporaryFile.virtualPath,
-              },
-            );
-            this.sftpConnection.status(
-              reqId,
-              SFTP_STATUS_CODE.FAILURE,
-              'An error occurred when attempting to register this file on Permanent.org.',
-            );
-          });
+    const serverResource = this.activeHandles.get(handle.toString());
+    if (serverResource === undefined) {
+      logger.verbose(
+        'Response: Status (FAILURE)',
+        {
+          reqId,
+          code: SFTP_STATUS_CODE.FAILURE,
         },
       );
+      this.sftpConnection.status(
+        reqId,
+        SFTP_STATUS_CODE.FAILURE,
+        'There is no server resource associated with this handle.',
+      );
       return;
     }
-    this.openFiles.delete(handle.toString());
-    this.openFilePaths.delete(handle.toString());
-    logger.verbose(
-      'Response: Status (OK)',
-      {
+    if (serverResource.resourceType === ServerResourceType.Directory) {
+      logger.verbose(
+        'Response: Status (FAILURE)',
+        {
+          reqId,
+          code: SFTP_STATUS_CODE.FAILURE,
+        },
+      );
+      this.sftpConnection.status(
         reqId,
-        code: SFTP_STATUS_CODE.OK,
-      },
-    );
-    this.sftpConnection.status(reqId, SFTP_STATUS_CODE.OK);
+        SFTP_STATUS_CODE.FAILURE,
+        'This handle does not refer to a file.',
+      );
+      return;
+    }
+    if (serverResource.resourceType === ServerResourceType.PermanentFile) {
+      this.activeHandles.delete(handle.toString());
+      logger.verbose(
+        'Response: Status (OK)',
+        {
+          reqId,
+          code: SFTP_STATUS_CODE.OK,
+        },
+      );
+      this.sftpConnection.status(reqId, SFTP_STATUS_CODE.OK);
+      return;
+    }
+    const { virtualFilePath } = serverResource;
+    (async () => {
+      let temporaryFile: TemporaryFile;
+      let fileSize: number;
+
+      try {
+        temporaryFile = await this.temporaryFileManager.getTemporaryFile(virtualFilePath);
+      } catch (err: unknown) {
+        logger.verbose(
+          'Response: Status (FAILURE)',
+          {
+            reqId,
+            code: SFTP_STATUS_CODE.FAILURE,
+            path: virtualFilePath,
+          },
+        );
+        this.sftpConnection.status(
+          reqId,
+          SFTP_STATUS_CODE.FAILURE,
+          'An error occurred when attempting to access the temporary file associated with this handle.',
```

Comment:
(picked this instance randomly but the comment applies everywhere)

The statuses are only sent to the client, and do not appear in the logs.  This is unfortunate and will make it harder to debug (the server logs just say there is a failure, but don't indicate the reason for the failure).

Reply:
(I don't think this should necessarily be addressed in this PR, since it applies to *all* failure messages in the project.)